src/main/scala/com/datastax/sparkstress/SparkCassandraStress.scala

```diff
@@ -155,63 +185,52 @@ object SparkCassandraStress {
         .setAppName("SparkStress_"+config.testName)
         //Make sure for streaming that the keep_alive is sufficently large
         .set("spark.cassandra.connection.keep_alive_ms", (config.streamingBatchIntervalSeconds*1000*5).toString)
+        .set("spark.cassandra.input.metrics", "true")
+        .set("spark.cassandra.output.metrics", "true")
         .setAll(config.sparkOps)
 
-    val sc = ConnectHelper.getContext(sparkConf)
-    
-    if (config.verboseOutput) { 
+    val ss = ConnectHelper.getSparkSession(sparkConf)
+
+
+    if (config.verboseOutput) {
       println("\nDumping debugging output")
-      println(sc.getConf.toDebugString+"\n")
+      println(ss.sparkContext.getConf.toDebugString+"\n")
     }
 
-    val test: StressTask =
-      config.testName.toLowerCase match {
-          /** Write Tasks **/
-        case "writeshortrow" => new WriteShortRow(config, sc)
-        case "writewiderow" => new WriteWideRow(config, sc)
-        case "writeperfrow" => new WritePerfRow(config, sc)
-        case "writerandomwiderow" => new WriteRandomWideRow(config, sc)
-        case "writewiderowbypartition" => new WriteWideRowByPartition(config, sc)
-
-          /** Read Tasks **/
-        case "pdcount" => new PDCount(config, sc)
-        case "ftsallcolumns" => new FTSAllColumns(config, sc)
-        case "ftsfivecolumns" => new FTSFiveColumns(config, sc)
-        case "ftsonecolumn" => new FTSOneColumn(config, sc)
-        case "ftspdclusteringallcolumns" => new FTSPDClusteringAllColumns(config, sc)
-        case "ftspdclusteringfivecolumns" => new FTSPDClusteringFiveColumns(config, sc)
-        case "jwcallcolumns" => new JWCAllColumns(config, sc)
-        case "jwcpdclusteringallcolumns" => new JWCPDClusteringAllColumns(config, sc)
-        case "jwcrpallcolumns" => new JWCRPAllColumns(config, sc)
-        case "retrievesinglepartition" => new RetrieveSinglePartition(config, sc)
-
-          /** Streaming Tasks **/
-        case "streamingwrite" => new StreamingWrite(config, sc)
-      }
+    val test: StressTask = {
+      val cl = getClass.getClassLoader
```

Comment:
not sure how i feel about this ... I tend to dislike reading things at runtime that we actually know at compile time. I think I would prefer we have this hardcoded, but we could just compose functions from both read and write tasks. So for example

Write tasks could have a functioN
paritalFunction [String] => StressTask
So inside the writer we list all the write tasks
Then we do the same thing for the reader

Then here we just do

val testMatchers = writeTestMatcher orElse ReadTestMatcher orElse throw TestNotFound or whatever
testMatchers(testName)

Reply:
This way we could be removing the code from this file and putting it where it belongs (in the test cases)

Reply:
Or if you want to do something real fun. _Looks around to make sure @pkolaczk isn't looking_ . You could make a Scala annotation that automatically generates the the list of all available tests at compile time. It doesn't count as a runtime exception if you do the reflection at compile time :D

Reply:
Yeah I wasn't a fan of this use of reflection either, I did this in response to a previous comment that asked for 
> maybe it would be easier to pickup all Task subclasses from the classpath and reflectively create an instance here?

So the first new suggestion here sounds like a non-reflection approach if I understand it correctly, which seems like a cleaner way to do it. I think I prefer that.